src/legacy/ui/public/vis/agg_config.ts

```diff
@@ -25,27 +25,39 @@
  */
 
 import _ from 'lodash';
-import { fieldFormats } from '../registry/field_formats';
+// @ts-ignore
 import { i18n } from '@kbn/i18n';
+import { AggType } from '../agg_types';
+// @ts-ignore
+import { fieldFormats } from '../registry/field_formats';
+import { writeParams } from '../agg_types/agg_params';
+import { AggConfigs } from './agg_configs';
+
+export interface AggConfigOptions {
+  id: string;
+  enabled: boolean;
+  type: string;
+  schema: string;
+  params: any;
+}
 
-class AggConfig {
-
+export class AggConfig {
   /**
    * Ensure that all of the objects in the list have ids, the objects
    * and list are modified by reference.
    *
    * @param  {array[object]} list - a list of objects, objects can be anything really
    * @return {array} - the list that was passed in
    */
-  static ensureIds(list) {
-    const have = [];
-    const haveNot = [];
-    list.forEach(function (obj) {
+  static ensureIds(list: AggConfig[]) {
+    const have = [] as any;
+    const haveNot = [] as any;
```

Comment:
Could these be `Array<Record<string, any>>`? Then you might not need the `any` on line 60?

Reply:
should be `AggConfig[]`